src/main/java/io/cryostat/discovery/CustomDiscovery.java

```diff
@@ -165,9 +161,7 @@ Response doV2Create(
 
             if (Target.find("connectUrl", target.connectUrl).singleResultOptional().isPresent()) {
                 return Response.status(Response.Status.BAD_REQUEST.getStatusCode())
-                        .entity(
-                                V2Response.json(
-                                        Response.Status.BAD_REQUEST, "Duplicate connection URL"))
+                        .entity("Duplicate connection URL")
```

Comment:
I think this `Response.status()` call can become `throw new BadRequestException("Duplicate connection URL")`, and similar for the next few lower down as well.